### PR TITLE
Fix Various Issues with Resource/Element Validation and Field Display

### PR DESCRIPTION
**Pull request by smg6511**
#Made several changes to gain expected validation behavior when editing resources, templates, tvs, snippets, chunks, and plugins. Now, the first tab with an invalid field is activated when not currently visible. Also small fixes to validation regex rules as well as message display were made. Lastly, some inconsistencies with field description labelling is addressed (i.e., when inline help is specified in system settings, all element editing fields now show their descriptions below instead of in tooltips). This labelling inconsistency should IMO be fixed in resource editing pages as well, but is not addressed here. (Please forgive all the minute space diffs; most of them are not intentional.)

### What does it do?

- Fixed regex in modx schema and class maps for element names (end of rules incorrectly had a negative _lookahead_ for space rather than negative _lookbehind_).
- Removed failureSubmit and traverseNode methods in MODx.panel.Resource. The showErroredTab and detectErrors methods in the parent MODx.FormPanel class, with adjustments, serve the same purpose. The failureSubmit listener of each element panel subclass now calls the showErroredTab method to evaluate field errors.
- Added helper method in MODx.FormPanel to convert an array of component itemIds into their associated component ids. This to ensure a proper value to pass to getCmp when a component's id is created on the fly.
- Added failure listener to element panels. Without it, the failureSubmit _event_ never gets triggered.
- Fixed element create and update processors to handle the name field for templates correctly. Without this fix, it was possible to save templates with duplicate names.
- Some general streamlining of defaults in element panel configs.
- The first-found (and now navigated to) field error, when being returned from the server side, would disappear as soon as the alert message appeared. This was due to the default behavior of native (Ext) validation triggering on blur. If the field in question is valid as far as the front end config rules go, its invalid state gets cleared once focus is lost even though the back end validation may still not be passing. Thus the default validation event was set to 'change' instead.

### Why is it needed?

To ensure better evaluation and display of, as well as dynamic navigation to, found field errors. At present, it can be confusing when encountering an error that is not in the currently active tab.

### Related issue(s)/PR(s)

Addresses issues #14743 , #15116 ,  and #15088

### CODE REVIEW

File: manager/assets/modext/widgets/resource/modx.panel.resource.js [hunk 1]
```diff
@@ -105,7 +121,7 @@ Ext.extend(MODx.panel.Resource,MODx.FormPanel,{
                 this.rteLoaded = false;
             }
         }
-
+        // console.log("MODx.panel.Resource::setup, isDirty?", this.isDirty());
```
Comment (Ruslan-Aleev) [inline manager/assets/modext/widgets/resource/modx.panel.resource.js]: This line should be deleted :)

Comment (smg6511) [reply manager/assets/modext/widgets/resource/modx.panel.resource.js]: Thought I'd tracked down all those logging lines! Fixed.

**Review by Ruslan-Aleev — APPROVED** [review]
**Great PR, thanks a lot!**
Solves a lot of problems in UX :)

The active sub-category of the TV tabs opens, and the screen "scrolls" to the error (when the TV position is below the content or in a separate tab):
tv_req_1

Messages are displayed and the tab is activated:
tv_req_2

**Comment by Ruslan-Aleev** [comment]
Guys, please test this PR, it significantly improves the current behavior for validation in forms.
It would be great to have this in the 2.8 branch :)

**Comment by Mark-H** [comment]
It looks our current js build workflow fails when using `let` and `const` constructs (in the uglifyjs bit) so I'm reverting those back to plain ol' `var`. Probably a good idea to sort out that workflow at some point, but for now I'm doing it this way to get your improvements into 2.8. Thanks @smg6511 for the great work.

**Review by Mark-H — APPROVED** [review]

**Comment by Ruslan-Aleev** [comment]
@Mark-H what about the 3.x branch? The 3.x branch also contains these bugs.

**Comment by Mark-H** [comment]
I'll try to cherry pick this (and the other things I merged the other day) into 3.x later. Couldn't get that to do what I wanted last sunday around midnight so figured that was a good one for another day. ;)

**Comment by smg6511** [comment]
Mark - If you want I can integrate the changes into a new PR for 3.x. Also, regarding your comment on the workflow you may want to take a look at Terser in place of Uglifyjs. I think ES6 has been out long enough and support is broad enough for us to be integrating its features into our coding.

**Comment by rthrash** [comment]
This pull request has been mentioned on **MODX Community**. There might be relevant details there:

https://community.modx.com/t/proposal-to-move-element-code-into-its-own-tab/4562/1

**Pull request merged**